karoo-gp.py

```diff
@@ -388,6 +389,172 @@
     mode = 's'
 
 
+#++++++++++++++++++++++++++++++++++++++++++
+#   Define pause callback                 |
+#++++++++++++++++++++++++++++++++++++++++++
+
+# used by: karoo-gp interactive
+def fx_karoo_pause_refer(model):
```

Comment:
I'm not sure these functions belong here.  Probably they should go in `pause.py` or `base_class.py`.